### ISSUE TITLE
Development use cases outline

### ISSUE THREAD

**Issue by kubaPod** [b3m2a1/mathematica-BTools#4]
# TL;DR;

It would be nice to have a short wiki with common development use cases / step and corresponding functions that achieve those goals. 

# Background

Very often I don't need golden hammer solutions but an access to those tiny features. I believe I'm not the only one as for anything more complex than a package there is hardly any widely established standard.

# Todo

So, in order to list those functions / lack of them we need to prepare a list of steps / use cases.

Here is a start, it is biased so needs to be refactored and extended before publishing.

## Creation / Development

- newProject(type)
- newProjectContent(type)  (*todo: expand*)
- createPacletInfo(initAssociationWithRequiredFields)
- mmaProjectGitInit() 
- createTests()

### PacletInfo:
- modifyPacletInfo(key:value) 
- handy PI getters
- incrementVersion(major|minor|patch : patch)
- addExtension(fe|docs|resource)

### Kernel:
- addSubpackage(packageTemplateType)

### FrontEnd
- addStylesheet()

### Documentation:
- newDocPage(sym|tutorial|guide|...)
- buildDocs()

### Tests:  
- runTests()

### Resources:
 ...

### Git:
- gitCommit(msg, type) => (add . + commit(msg, type)) 
     // by type I assume (in future) support for something like https://conventionalcommits.org/spec/v1.0.0-beta.1.html

## Release / Deployment

- gitTag()  // at least `git tag -a -m {most recent feat(_) commit msg} v{paclet version}`
- release as github asset

The idea is to outline more than it is to know which features need care.

**Comment by b3m2a1** [comment]
Most of this is already built in. I will attempt to document these things sometime in the coming two weeks.

**Comment by kubaPod** [comment]
Thanks @b3m2a1 , keep in mind that a full symbols documentation is not essential, only a single tutorial-like page with usage signature for each use case will be enough.

**Comment by b3m2a1** [comment]
@kubaPod I got a quick thing up https://github.com/b3m2a1/mathematica-BTools/blob/master/project/examples/BasicUsage.md

**Comment by kubaPod** [comment]
@b3m2a1  

1. few images in documentation section of BasicUsage.md are 404 now
2. how to update e.g. a symbol page which already contains manually created examples? I guess one can add it manually or just generate a symbol page again and copy paste from previous version. Either way folks may be wondering what is the way and if they are not missing something. Worth to state that clearly, what do you think?

**Comment by kubaPod** [comment]
4. Is it possible to use DocGen to create and build documentation without all this configuration steps required for loading a building the project in general? I'm trying to add documentation to an existing package, DocGen works fine to generate SymbolFile templates but I am having troubles to build(with 
indexing)documentation from desired source directory to desired build directory.

**Comment by b3m2a1** [comment]
@kubaPod I moved the document to a Wiki: https://github.com/b3m2a1/mathematica-BTools/wiki/BasicUsage. See also https://github.com/b3m2a1/mathematica-BTools/wiki/Documentation.

This should fix (1)

I don't have a super clean workflow for (3), yet, but I can write a document explaining exactly what one would have to do.

For (2) I think you can use the template generated by `DocGen["SymbolPage", test, Method->"Template"]` to make a notebook where you can make new examples to link in. In general, though, I save my templates for future editing. I can make that clearer though.

For (4) this already is possible (as I think that document shows). I'll edit that more when I have time, though.

All these notes will end up on the wiki soon enough

**Comment by kubaPod** [comment]
@b3m2a1  Don't know if this is a proper place to point minor problems. Let me know if not.

Would be nice to add ``Needs @  "BTools`External`"`` somewhere around GitHub section of BasicUsage.

**Comment by b3m2a1** [comment]
@kubaPod that's [here](https://github.com/b3m2a1/mathematica-BTools/wiki/GitHub#general-form) now. Haven't merged it into the main usage cases doc but could.

**Comment by kubaPod** [comment]
@b3m2a1  I think it can be closed. If I come with more specific issues I will create them.

**Comment by b3m2a1** [comment]
@kubaPod sounds good. I think it can be reopened later, too, if need be.
